### PR TITLE
Merge upstream changes

### PR DESCRIPTION
**Pull request by caesar2164**
@stvstnfrd this is a straight cherry-pick of each commit on edX's master missing from our master...

### CODE REVIEW

File: gen_cert.py [hunk 1]
```diff
@@ -138,8 +138,7 @@ def __init__(
                            template_pdf parameter
         """
         if dir_prefix is None:
-            if not os.path.exists(TMP_GEN_DIR):
-                os.makedirs(TMP_GEN_DIR)
+            self._ensure_dir(TMP_GEN_DIR)
```
Comment (caesar2164) [inline gen_cert.py:141]: this function does exactly what is replaced, so seems like a good change

File: .travis.yml [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
   - "2.7"
 env:
   - BOTO_CONFIG=/dev/null
+branches:
+  only:
+    - master
 before_install:
-  - pip install pep8
-  - pep8 --verbose
+  - export BOTO_CONFIG=/dev/null
```
Comment (caesar2164) [inline .travis.yml]: we have this in `env:`, which is better?

Comment (stvstnfrd) [reply .travis.yml]: Probably our way? [1]

Which commit is older? Ours or theirs?

- [1] https://docs.travis-ci.com/user/environment-variables/

Comment (caesar2164) [reply .travis.yml]: ours is 8 months old, edX's 6 months old

Comment (stvstnfrd) [reply .travis.yml]: Ours seems more correct, since it's prescribed by Travis-CI;
lets use it.

**Review by stvstnfrd — APPROVED** [review]

**Pull request merged**